### PR TITLE
Feat/implement endpoint

### PR DESCRIPTION
**Pull request by Devynnem**
____ Wrote Tests ____ Implemented ____ Reviewed

## Neccesary checkmarks
- [ ] All Tests are Passing in all environments
- [ ] The code will run locally
## Type of change
- [x] New feature
- [ ] Bug Fix
- [x] Refactor
## Description of Change:

  description: Added a apiCalls fill to store our fetch call, added some favorite functionality to allow us to remove a favorite truck while on the favorites page. However, it is removing all of the vendors when you unclick a favorite from the main page.
    
## Requested feedback: Favorites functionality and then truck details fetch call with our new endpoint from backend
    
  I'd like feedback on... 

## Check the correct boxes

- [ ] This broke nothing
- [x] This broke some stuff
- [ ] This broke everything

## Testing Changes

- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [ ] My code has no unused/commented out code
- [ ] My code has no binding.pry's
- [ ] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: src/utilities/socket.js [hunk 1]
```diff
@@ -3,6 +3,6 @@ import { io } from 'socket.io-client'
 export const socket = io("http://localhost:3001", {
```
Comment (Pma913) [inline src/utilities/socket.js:3]: Glad it worked placing the socket connection in its own file!

**Review by emlindvall — APPROVED** [review]
Super, thank you Devynne!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -55,9 +49,7 @@ const App = () => {
 // }
 
   const removeFav =  (truck) => {
-
-  const newFavState = favorites.filter((fav) => fav.id !== truck.id)
-  setFavorites(newFavState)
+    setVendors(vendors.filter((v) => v.favorited === true))
```
Comment (Pma913) [inline src/Components/App/App.js:52]: This is nice for when we are in favorites view. Will be interesting to make it not effect the other vendors when in the main page.

**Review by Pma913 — APPROVED** [review]
looks good. Some more cleanups but a nice start

**Pull request merged**